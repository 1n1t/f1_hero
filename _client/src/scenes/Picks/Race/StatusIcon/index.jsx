import React from 'react';
import PropTypes from 'prop-types';

const StatusIcon = ({ iconClass, status }) => (
	<i
		className={`fa ${iconClass}`}
		aria-hidden="true"
		data-test={`race-status-${status}`}
	/>
);

StatusIcon.propTypes = {
	iconClass: PropTypes.string.isRequired,
	status: PropTypes.oneOf(['new', 'submitted', 'passed']).isRequired
};

export default StatusIcon;
