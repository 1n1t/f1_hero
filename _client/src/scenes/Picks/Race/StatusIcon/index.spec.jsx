import r from 'Root/jest/utils/renderer';

import StatusIcon from './index';

const defaultProps = {
	iconClass: 'fa-check',
	status: 'passed'
};

r.init(StatusIcon, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render();
	expect(wrapper).toMatchSnapshot();
});
