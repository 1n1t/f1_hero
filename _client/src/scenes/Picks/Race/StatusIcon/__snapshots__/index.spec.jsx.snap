// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<i
  aria-hidden="true"
  className="fa fa-check"
  data-test="race-status-passed"
/>
`;
